sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -225,6 +231,37 @@ public int getSize(Message amqpMessage) {
         return amqpMessage;
     }
 
+    private Map<String, Object> convertToDescribedType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof URI) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.URI_SYMBOL, ((URI) value).toString()));
+            } else if (value instanceof OffsetDateTime) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.OFFSETDATETIME_SYMBOL, ((OffsetDateTime) value).atZoneSameInstant(ZoneOffset.UTC).toInstant().toEpochMilli()));
+            } else if (value instanceof Duration) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.DURATION_SYMBOL, ((Duration) value).toNanos()));
+            }
+        }
+        return propertiesValue;
+    }
+
+    private Map<String, Object> convertToOriginType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof DescribedType) {
+                DescribedType describedType = (DescribedType) value;
+                if (ServiceBusConstants.URI_SYMBOL.equals(describedType.getDescriptor())) {
+                    propertiesValue.put(entry.getKey(), URI.create((String) describedType.getDescribed()));
```

Comment:
Do we need to do error handling for this kind of deserialization? `URI.create((String) describedType.getDescribed())`

Reply:
Agree. It could have been sent from a client that might have different validation on URI, or not even validate on it as it basically just a String.